### ISSUE TITLE
When do pseudo elements become accessible to developer styles?

### ISSUE THREAD

**Issue by khushalsagar** [WICG/shared-element-transitions#106]
The transition progresses in the following stages :

- Prepare : Cache state from the old DOM asynchronously.
- PrepareResolved : Prepare finishes and developer is notified to switch to the next DOM. Rendering is suppressed at this stage.
- Start : The new DOM has loaded enough that the transition can be started.
- StartFinished : All animations for a transition have been finished.

The pseudo elements for the transition DOM are generated by the browser when the prepare phase of the transition starts. This makes it easy to cache state from the old DOM. But it's not clear whether applying developer styles to these elements should be allowed at this stage.

The main consideration here is that the developer styles will likely have animations so applying them sooner than the start stage would start the animations at an incorrect time. So the simplest option is to defer applying developer styles until the start stage.

**Comment by vmpstr** [comment]
I feel like that we shouldn't be creating a pseudo element structures at the prepare phase. As you say, this may cause developer style sheets to take effect which is undesirable and I would like to avoid doing special casing like "yes, the style _would_ apply, but this is a SET pseudo element in prepare phase, so don't". 

I think the proper course of action here is to not create pseudo element hierarchy for baked snapshot captures. In other words, the first time we would create the pseudo element structure is at the start phase, which would create pseudos that both reference the captured snapshot and the new live snapshot and developer style sheets apply as typical.

The question is how do we capture the baked snapshot, but I think there are ways to do that without creating pseudo element structures

**Comment by khushalsagar** [comment]
Vlad and I discussed this offline. The key point was about what effect creating the pseudo element structure earlier has on input and rendering.

Does input processing work the same way as described in #88 during the prepare phase then? This implies that interactivity would pause similar to what happens when the transition starts. This is probably okay given that we have to suppress rendering during this time anyway. I filed a separate [issue](https://github.com/WICG/shared-element-transitions/issues/110) to clarify the details of suppressing rendering.

@vmpstr was there anything else I might have missed. The conclusion was that it's okay for the partial pseudo element tree (which doesn't include the image pseudo element for new content) to be created and exposed during the prepare phase.
